src/navigation_controller.ts

```diff
@@ -950,6 +950,37 @@ export class NavigationController {
     ContextMenuRegistry.registry.register(copyAction);
   }
 
+  /**
+   * Register the action for inserting above a block.
+   */
+  protected registerInsertAction() {
+    const insertAboveAction: ContextMenuRegistry.RegistryItem = {
+      displayText: (scope) => 'Keyboard Navigation: insert above',
+      preconditionFn: (scope) => {
+        const ws = scope.block?.workspace;
+        if (!ws) return 'hidden';
+
+        if (!scope.block?.previousConnection) return 'hidden';
```

Comment:
Is it disabled on the top block of the stack because tryToConnectNodes can't handle that case yet?

Reply:
This is the case where there's no previous connection, so there's no way to insert above. This comes up for the event blocks.